src/services/pagination/LengthAwarePaginator.js

```diff
@@ -0,0 +1,81 @@
+import Paginator from './Paginator';
+import LesgoException from '../../exceptions/LesgoException';
+
+const FILE = 'Services/pagination/LengthAwarePaginator';
+
+export default class LengthAwarePaginator extends Paginator {
+  /**
+   * Constructor
+   *
+   * @param db
+   * @param sql
+   * @param sqlParams
+   * @param perPage
+   * @param currentPage
+   * @param total
+   */
+  constructor(db, sql, sqlParams, perPage = 10, currentPage = 1, total = null) {
+    if (total !== null && typeof total !== 'number') {
+      throw new LesgoException(
+        "Invalid type for 'total'",
+        `${FILE}::INVALID_TYPE_TOTAL`,
+        500,
+        { perPage }
+      );
+    }
+    super(db, sql, sqlParams, perPage, currentPage);
+    this.totalProp = total;
+  }
+
+  /**
+   * Get the last page.
+   *
+   * @returns {Promise<number>}
+   */
+  async lastPage() {
+    return this.calculateTotalNumberOfPages();
+  }
+
+  /**
+   * Total items in all pages.
+   *
+   * @returns {null|number}
+   */
+  async total() {
+    if (this.totalProp === null) {
+      this.totalProp = await this.countTotalItems();
```

Comment:
Can be simplified as 

return this.totalProp || await this.countTotalItems()

Reply:
Thanks @chocnut